### PR TITLE
Refactor EvolveAll and InitialTransfer workers

### PR DESCRIPTION
**Pull request by TheSavior**
Moving more things into the worker so that the initial tick is clearer and flatter.

### CODE REVIEW

**Comment by douglascamata** [comment]
@SteffanLong how is your work towards the `ReleaseWorker`?

File: pokemongo_bot/cell_workers/evolve_all_worker.py [hunk 1]
```diff
@@ -48,6 +51,35 @@ def work(self):
                 ))
                 self._release_evolved(release_cand_list_ids)
 
+    def _should_run(self):
+        # Will skip evolving if user wants to use an egg and there is none
+        skip_evolves = False
+
+        # Pop lucky egg before evolving to maximize xp gain
+        use_lucky_egg = self.config.use_lucky_egg
+        lucky_egg_count = self.bot.item_inventory_count(Item.ITEM_LUCKY_EGG.value)
```
Comment (TyrannosaurusREKT) [inline pokemongo_bot/cell_workers/evolve_all_worker.py]: You're missing the `Item` import in this module

Comment (TheSavior) [inline pokemongo_bot/cell_workers/evolve_all_worker.py]: Great catch!

**Comment by douglascamata** [comment]
Before merging this, we should merge this one: https://github.com/PokemonGoF/PokemonGo-Bot/pull/1023

**Comment by douglascamata** [comment]
LGTM!

**Pull request merged**